clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/images/memory/FileMemoryAccessObject.kt

```diff
@@ -0,0 +1,64 @@
+package com.clevertap.android.sdk.inapp.images.memory
+
+import com.clevertap.android.sdk.inapp.images.memory.MemoryDataTransformationType.MEMORY_DATA_TRANSFORM_TO_BITMAP
+import com.clevertap.android.sdk.inapp.images.memory.MemoryDataTransformationType.MEMORY_DATA_TRANSFORM_TO_BYTEARRAY
+import com.clevertap.android.sdk.inapp.images.memory.MemoryDataTransformationType.MEMORY_DATA_TRANSFORM_TO_FILE
+import com.clevertap.android.sdk.utils.CTCaches
+import java.io.File
+
+class FileMemoryAccessObject(private val ctCaches: CTCaches): MemoryAccessObject<ByteArray> {
+
+    override fun fetchInMemory(key: String): Pair<ByteArray, File>? {
+        val fileInMemory = ctCaches.fileLruCache()
+        return fileInMemory.get(key)
+    }
+
+    override fun fetchInMemoryAndTransform(key: String, transformTo: MemoryDataTransformationType): Any? {
+        val pair = fetchInMemory(key)
+        return pair?.let {
+            when(transformTo)
+            {
+                MEMORY_DATA_TRANSFORM_TO_BITMAP -> bytesToBitmap(it.first)
+                MEMORY_DATA_TRANSFORM_TO_BYTEARRAY -> it.first
+                MEMORY_DATA_TRANSFORM_TO_FILE -> it.second
+            }
+        }
+    }
+
+    override fun fetchDiskMemoryAndTransform(key: String, transformTo: MemoryDataTransformationType): Any? {
```

Comment:
can we have specific return type

Reply:
requires generics to reduce duplicate code and functions